configure.ac

```diff
@@ -301,6 +301,7 @@ AC_CHECK_FUNCS( futimes )
 AC_CHECK_FUNCS( wcslcpy lrand48_r killpg )
 AC_CHECK_FUNCS( backtrace_symbols getifaddrs )
 AC_CHECK_FUNCS( futimens clock_gettime )
+AC_CHECK_FUNCS( localtime_r )
```

Comment:
Somewhat surprisingly we don't currently have a single `localtime()` call in the code so it's use in this change would be thread-safe since it is the only thread that would ever call it. However, that's putting down a land mine. So this is a case where a fallback function needs to be defined if the platform doesn't provide it. The fallback should just use a simple global lock around the call to `localtime()` to ensure the single internal result buffer isn't overwritten before it's contents are copied into the user supplied result buffer. Look in the code for instances of `scoped_lock locker(lock);  //!OCLINT(side-effect)` for how to do that.

I should also note that you change has another bug: You don't call `tzset()` (and neither does the current fish code). It's possible the internal timezone state is being set indirectly via some other call but you shouldn't rely on that.